### PR TITLE
(feat) signin

### PR DESCRIPTION
**Pull request by underaj**

### CODE REVIEW

File: src/signin/Signin.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import React, { Component } from 'react';
+import {
+  StyleSheet,
+  Text,
+  View,
+  TouchableHighlight,
+} from 'react-native';
+import * as actionCreators from './signinActions';
+
+const t = require('tcomb-form-native');
```
Comment (dting) [inline src/signin/Signin.js:13]: Can this be changed to

```
import t from 'tcomb-form-native';
```

?

Comment (underaj) [inline src/signin/Signin.js:13]: done

File: src/signin/Signin.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+  signIn() {
+    const value = this.refs.form.getValue();
+    if (value) { // if validation fails, value will be null
+      this.props.actions.loginUserRequest();
+    }
+  }
+
+  render() {
+    let statusMessage;
```
Comment (dting) [inline src/signin/Signin.js:95]: We could use

```
const statusMessage = this.props.auth.statusMessage;
```

here and then on line 117:

```
{statusMessage && <Text style={styles.notification}>{statusMessage}</Text>}
```

Comment (underaj) [inline src/signin/Signin.js:95]: neat, done

File: src/signin/Signin.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    fontSize: 15,
+    alignSelf: 'center',
+    marginBottom: 30,
+  },
+});
+
+class SignIn extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```
Comment (dting) [inline src/signin/Signin.js:82]: Where is this state used? Should this component have a local state?

Comment (underaj) [inline src/signin/Signin.js:82]: The state was for the status message, taking it out now

File: src/signin/signinReducers.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const initialState = {
+  token: null,
+  userName: null,
+  isAuthenticated: false,
+  isAuthenticating: false,
+  statusMessage: null,
+};
+
+export default function (state = initialState, action) {
```
Comment (dting) [inline src/signin/signinReducers.js:9]: Does this pass the linter? I thought airbnb required names for functions expressions.

**Comment by dting** [comment]
LGTM thanks.

File: src/signin/signinReducers.js (hunk 4 above)
Comment (underaj) [inline src/signin/signinReducers.js:9]: passed, using react-native linters

Comment (dting) [inline src/signin/signinReducers.js:9]: cool

**Pull request merged**